fish-rust/src/wcstringutil.rs

```diff
@@ -248,7 +248,10 @@ pub fn string_fuzzy_match_string(
 /// This invokes \p func with (const char*, size_t) pairs.
 /// If \p func returns false, it stops; otherwise it continues.
 /// \return false if the callback returned false, otherwise true.
-pub fn wcs2string_callback(input: &wstr, mut func: impl FnMut(&[u8]) -> bool) -> bool {
+pub fn wcs2string_callback<IntoChars>(input: IntoChars, mut func: impl FnMut(&[u8]) -> bool) -> bool
```

Comment:
This should only be called with stuff that was either a wide string literal, or came from `str2wcstring`. Any other argument is likely a bug; therefore this should continue to take `&wstr`.